modin/pandas/test/dataframe/test_default.py

```diff
@@ -228,19 +228,107 @@ def test_combine_first():
-    eval_general(
-        *create_test_dfs(test_data["float_nan_data"]),
-        lambda df: df.corr(min_periods=min_periods),
-        comparator=modin_df_almost_equals_pandas,
+@pytest.mark.skipif(
+    Engine.get() == "Python",
+    reason="FIXME ASAP: https://github.com/modin-project/modin/issues/5916",
+)
+class TestCorr:
+    def test_corr(self):
+        eval_general(
+            *create_test_dfs(test_data["int_data"]),
+            lambda df: df.corr(),
+        )
+        # Modin result may slightly differ from pandas result
+        # due to floating pointing arithmetic.
+        eval_general(
+            *create_test_dfs(test_data["float_nan_data"]),
+            lambda df: df.corr(),
+            comparator=modin_df_almost_equals_pandas,
+        )
+
+    @pytest.mark.parametrize("min_periods", [1, 3, 5, 6])
+    def test_corr_min_periods(self, min_periods):
+        # only 3 valid values
+        eval_general(
+            *create_test_dfs({"a": [1, 2, 3], "b": [3, 1, 5]}),
+            lambda df: df.corr(min_periods=min_periods),
+        )
+
+        # only 5 valid values
+        eval_general(
+            *create_test_dfs(
+                {"a": [1, 2, 3, 4, 5, np.nan], "b": [1, 2, 1, 4, 5, np.nan]}
+            ),
+            lambda df: df.corr(min_periods=min_periods),
+        )
+
+        # only 4 valid values
+        eval_general(
+            *create_test_dfs(
+                {"a": [1, np.nan, 3, 4, 5, 6], "b": [1, 2, 1, 4, 5, np.nan]}
+            ),
+            lambda df: df.corr(min_periods=min_periods),
+        )
+
+        if StorageFormat.get() == "Pandas":
+            # only 4 valid values located in different partitions
+            modin_df, pandas_df = create_test_dfs(
+                {"a": [1, np.nan, 3, 4, 5, 6], "b": [1, 2, 1, 4, 5, np.nan]}
+            )
+            modin_df = pd.concat([modin_df.iloc[:3], modin_df.iloc[3:]])
+
+            assert modin_df._query_compiler._modin_frame._partitions.shape == (2, 1)
+            eval_general(
+                modin_df, pandas_df, lambda df: df.corr(min_periods=min_periods)
+            )
+
+    @pytest.mark.parametrize("numeric_only", [True, False, None])
+    def test_corr_non_numeric(self, numeric_only):
+        eval_general(
+            *create_test_dfs({"a": [1, 2, 3], "b": [3, 2, 5], "c": ["a", "b", "c"]}),
+            lambda df: df.corr(numeric_only=numeric_only),
+        )
+
+    @pytest.mark.skipif(
+        StorageFormat.get() != "Pandas",
+        reason="doesn't make sense for non-partitioned executions",
     )
+    def test_corr_nans_in_different_partitions(self):
+        # NaN in the first partition
+        modin_df, pandas_df = create_test_dfs(
+            {"a": [np.nan, 2, 3, 4, 5, 6], "b": [3, 4, 2, 0, 7, 8]}
+        )
+        modin_df = pd.concat([modin_df.iloc[:2], modin_df.iloc[2:4], modin_df.iloc[4:]])
+
+        assert modin_df._query_compiler._modin_frame._partitions.shape == (3, 1)
```

Comment:
Why do we want to go into query compiler, then modin frame level, and then partitions? feels too deep, plus is implementation-specific, not every query compiler would have a frame...

Reply:
that's why there's a `skipif` above this test making it to run only when Pandas storage format is specified, as in my understanding `PandasQueryCompiler` always have a frame, doesn't it?